### PR TITLE
Pass the file name of the linted file to flake8

### LINKED ISSUE #174: flake8 unaware of the filename of the scanned file
currently, `linter-flake8` lints the file by piping the file content to `flake8` stdin, with a command line like:

```
flake8 --max-complexity -1 --config /path/to/you/project/.flake8 -
```

(depending on your configuration, i.e. if it has `.flake8` set as `Project Config File`).

However, this means that rules about file names stored int the configuration file are not taken into account (i.e., settings like `exclude` and `filename`).

In my particular case, I'm using [flake8-putty](https://github.com/jayvdb/flake8-putty) and the stdin-feeding is causing the `putty-ignore` rules with file patterns to be skipped.

I understand that `stdin` is used because we want to lint the unsaved contents of the file, and unfortunately I haven't found any `flake8` switch for specifying what is the name of the file piped in the stdin, so this issue might not be able to be solved in the general case.

But if the file is saved, flake8 should be invoked on the actual file in the filesystem to avoid this issue.

### PR DESCRIPTION
**Pull request by leorochael**
This issue fixes #174, but depends upon acceptance of [this pep8 PR](https://github.com/PyCQA/pycodestyle/pull/493).

### CODE REVIEW

File: lib/main.coffee [hunk 1]
```diff
@@ -195,6 +195,7 @@ module.exports =
           parameters.push('--select', selectErrors.join(','))
         if (projectConfigFile = atom.config.get('linter-flake8.projectConfigFile'))
           parameters.push('--config', path.join(atom.project.getPaths()[0], projectConfigFile))
+        parameters.push("--stdin-name", filePath)
```
Comment (Arcanemagus) [inline lib/main.coffee]: Different quote style from the rest of the file it looks like.

Comment (leorochael) [inline lib/main.coffee]: Got it!

**Comment by Arcanemagus** [comment]
Marking this as closed, if and/or when `pep8` brings in support for that (and `flake8` updates to a supported version!) we can re-open this.

**Comment by Arcanemagus** [comment]
Hmmm, I _just_ saw your message when I was about to refer to this PR as to why it wasn't in yet. If you want to send that in that would be great! Otherwise I'll get to it eventually.

**Pull request closed without merging**